modules/generator/generator.go

```diff
@@ -38,6 +38,12 @@ const (
 	// We use a safe default instead of exposing to config option to the user
 	// in order to simplify the config.
 	ringNumTokens = 256
+
+	// NoGenerateMetricsContextKey is used in request contexts/headers to signal to
+	// the metrics generator that it should not generate metrics for the spans
+	// contained in the requests. This is intended to be used by clients that send
+	// requests for which span metrics have already been generated elsewhere.
```

Comment:
This will also skip generating service graph metrics:

```suggestion
	// contained in the requests. This is intended to be used by clients that send
	// requests for which span and service graph metrics have already been generated elsewhere.
```

Reply:
I was thinking of "span metrics" as a general term for all metrics derived from spans (so including service graph metrics), but I see how this is confusing with the terminology that already exists. Will change.